server/controllers/testsController.js

```diff
@@ -0,0 +1,193 @@
+    return next({
+      log: 'verifyInput middleware error: ' + err,
+      message: { err }
+    });
+  }
+};
+
+/* Converts header part of input object into header of test code
+
+  EXPECTED INPUT: res.locals.header, res.locals.assertions
+  EXPECTED OUTPUT: res.locals.headerOutput = Array of Strings
+    [
+      `describe('/', () => {`,
+        `describe('GET', () => {`,
+          `it('responds with status 200 and content type text/html', () => request(server)`,
+            `.get('/')`
+      ]
+*/
+testsController.headerGenerator = (header, assertions) => {
+  const headerOutput = [];
+
+  let description = '';
+
+  for (let assertion of assertions) {
+    if (Object.keys(assertion)[0] === 'status') {
+      description += description ? ' and ' : '';
+      description += `status ${assertion.status}`;
+    }
+    if (Object.keys(assertion)[0] === 'content') {
+      description += description ? ' and ' : '';
+      description += `content-type ${assertion.content}`;
+    }
+  }
+
+  description = `responds with ` + description;
+  if (!assertions.length)
+    description = `makes a ${header.method} request to "${header.endpoint}"`;
+
+  headerOutput.push(`describe('${header.endpoint}', () => {`);
+  headerOutput.push(` describe('${header.method}', () => {`);
+  headerOutput.push(` it('${description}', () => request(server)`);
+  headerOutput.push(`  .${header.method.toLowerCase()}('${header.endpoint}')`);
+
+  if (
+    (header.method === 'POST' ||
+      header.method === 'PATCH' ||
+      header.method === 'DELETE') &&
+    header.req_body
+  ) {
+    headerOutput.push(`   .send(${header.req_body})`);
+  }
+
+  return headerOutput;
+};
+
+testsController.createHeaderText = (req, res, next) => {
+  const header = res.locals.header;
+  const assertions = res.locals.assertions;
+
+  res.locals.headerOutput = testsController.headerGenerator(header, assertions);
+
+  return next();
+};
+
+//Maybe need to write a middleware to reorganize the res array?
+/* 
+  INPUT: assertions: [
+                    { status: 200 },
+                    { content: 'text/html' },
+                    { res_body: { a: 'b' } }
+      ]
+*/
+//Write a MiddleWare to examine the res object from the front-end and produce lines of code depending on what is coming in from the middleware.
+testsController.middleGenerator = (assertions) => {
+  let middleOutput = [];
+  //the it statement: will need some logic there when generating the string to be returned.
+  //iterate through the array and push a string that will need to be typed into an output array.
+  //the get should be simple enough. For this type of testing I think it will just be to the same route as the header route will be.
+  for (let assertion of assertions) {
+    //So if the dropdown here is status/content/body we push a different string into the output array.
+    //how to access this dropdown item? res.locals.data.
+    // console.log(assertion);
+    if (Object.keys(assertion)[0] === 'status') {
```

Comment:
Would storing `Object.keys(assertion)` as a variable help to keep code DRY and reduce time complexity?